src/effects/Tween.js

```diff
@@ -27,12 +27,17 @@ Tween.prototype = {
 			Tween.propHooks._default.get( this );
 	},
 	run: function( percent ) {
-		var hooks = Tween.propHooks[ this.prop ];
+		var eased,
+			hooks = Tween.propHooks[ this.prop ];
 
-		this.pos = this.options.duration ?
-			jQuery.easing[ this.easing ]( percent ) :
-			percent;
-		this.now = ( this.end - this.start ) * this.pos + this.start;
+		if ( this.options.duration ) {
+			this.pos = eased = jQuery.easing[ this.easing ](
+				percent, this.options.duration * percent, 0, 1, this.options.duration
+			);
```

Comment:
The main difference is the addition of the `eased` parameter that seems redundant here. I'd check the size impact of that.